### DISCUSSION TITLE
Cannot Debug

### DISCUSSION THREAD

**Discussion by Ebaed135** [microsoft/vscode-discussions#758] [Extension Development QnA] [answered]
Hello everyone, I am new to coding and I am getting started  by watching a Youtube video by Bro Code, C programming full course. I followed his steps downloading VS code and setting everything up the way he explained. However, when I went to write my first piece of code I again followed what he typed, but when I attempt to run the code it does not work. A pop up on the bottom right says that it cannot build or debug because the active file is not a C or C++ source file. I tried to troubleshoot but I really could not find anything to solve it. One thing that I did I notice was that under my run code button there is a bug, which is not present in the video. I would really appreciate any help to solve this problem. I left a screen shot of the screen below. Furthermore, let me know if there is any other information  or screen shots I need to give to help find a solution. Thank you!

<img width="1470" alt="Screenshot 2023-08-14 at 9 53 57 PM" src="https://github.com/microsoft/vscode-discussions/assets/142280268/32187f85-1f40-4938-b196-1198cf36198e">

**Comment by alefragnani** [comment, marked as answer]
Hi @Ebaed135 ,

Based on your specific question, as you are following the steps of a YouTube video tutorial, I think the best location to get insights about your problem is it its comments section. The author must be very familiar with issues like yours, and should be able to help you easily. And if the issue is related with the C/C++ extension instead, the author would probably send you to the extension's repo.

It's been a long time I don't work with C/C++, so I may be a little rusty, but it seems you have some issue with the extension itself, not the C/C++ compiler. I'm not sure the extension requires you have focus on the file, or use context menu, but worth a try. Maybe the extension's repo would be a good place to find answers as well.

By the way, for future feature questions about the C/C++ extension usage, please use its repo (https://github.com/Microsoft/vscode-cpptools). This Discussions repo is focused for extension development for now.

Hope this helps
